src/tokens/saga.ts

```diff
@@ -309,6 +312,62 @@ export function* watchAccountFundedOrLiquidated() {
   }
 }
 
+export function* fetchImportedTokensBalances(address: Address) {
+  try {
+    const showImportedTokens = yield* call(
+      getFeatureGate,
+      StatsigFeatureGates.SHOW_IMPORT_TOKENS_FLOW
+    )
+    if (!showImportedTokens) {
+      return {}
+    }
+
+    const importedTokens: StoredTokenBalances = yield* select(importedTokensInfoSelector)
+    const importedTokensList = Object.values(importedTokens)
+
+    const balanceRequests: any = {}
+    importedTokensList.forEach((importedToken) => {
+      if (!importedToken) {
+        return
+      }
+
+      const contract = getContract({
+        abi: erc20.abi,
+        address: importedToken!.address as Address,
+        client: {
+          public: publicClient[networkIdToNetwork[importedToken.networkId]],
+        },
+      })
+
+      balanceRequests[importedToken.tokenId] = contract.read.balanceOf([address])
+    })
+
+    const balances = (yield* all(balanceRequests)) as Record<string, bigint>
```

Comment:
I have tried to execute all these requests using the multicall method, but I can't find why it's not working.
I got some `undefined is not a function` error that I can't understand where is coming from (it seems like inside the multicall call).
I think this is fine if the user only have a few imported tokens.